torchtune/modules/optim/_cpu_offload.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torchtune.config._utils import _get_component_from_path
+
+try:
+    from torchao.prototype.low_bit_optim import (
+        CPUOffloadOptimizer as _CPUOffloadOptimizer,
+    )
+
+    _CPU_OFFLOAD_OPTIM_AVAILABLE = True
+
+    class CPUOffloadOptimizer(_CPUOffloadOptimizer):
+        """Offload optimizer to CPU for single-GPU training. This will reduce GPU memory by the size of optimizer
+        state. Optimizer step will be done on CPU.
+
+        Args
+            params: a list of parameters or parameter groups.
+            optimizer_class: constructor of the base optimizer.
```

Comment:
Adding an example here would be great. E.g. 'torch.optim.AdamW', so its clear for the user how to use it. Maybe even the an instantiation example would be nice, like:

```
Example:
	from torchtune.modules.optim import CPUOffloadOptimizer
	optimizer=CPUOffloadOptimizer(
		params=model.params(), 
		optimizer_class='torch.optim.AdamW', 
		offload_gradients=False, 
		lr=2e-5
	)